### PR TITLE
Add team assignments module

### PR DESCRIPTION
**Pull request by embarnard**
## Link to pivotal/JIRA issue
- e.g. [link](https://codeforamerica.atlassian.net/browse/GYR1-501?atlOrigin=eyJpIjoiMDc4NjNjNDAyNWZkNDc3ZjkyNTUzMDU4YjVjZTMzYjYiLCJwIjoiaiJ9)
## Is PM acceptance required? (delete one)
- Yes - don't merge until JIRA issue is accepted!
## What was done?
- Added a "team assignments" module to the hub dashboard
- This module is used by organization leads and site coordinators to be able to view other users and how many assigned tax returns they have
## How to test?
  - test in heroku, login as an org lead or a site coordinator ("cucumber@example.com" or "skywalker@example.com" logins)
  - this view will not show up for coalition leads or other users
- Risk Assessment
  - check that only displays accessible users 
## Screenshots (for visual changes)
- Before
Screenshot 2024-07-31 at 12 50 26 PM

- After
Screenshot 2024-07-30 at 8 35 28 PM

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -241,6 +241,27 @@ def state_file_admin?
...
+      "Admin"
+    when "ClientSuccessRole"
+      "Client Success"
+    when "GreeterRole"
+      "Greeter"
+    else
+      ""
+    end
+  end
+
```
Comment (tofarr) [inline app/models/user.rb:264]: Did this get pulled in from Kevin's PR? Should this be internationalized?

Comment (embarnard) [reply app/models/user.rb:264]: No it didn't, did he do something similar? I can rebase and about internationalization, good point it should

File: app/presenters/hub/dashboard/team_assignment_presenter.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+                                   elsif @current_user.site_coordinator?
+                                     sites = @current_user.role.sites
+                                     site_coordinators = User.where(role: SiteCoordinatorRole.assignable_to_sites(sites))
+                                     team_members = User.where(role: TeamMemberRole.assignable_to_sites(sites))
+                                     site_coordinators.or(team_members)
+                                   end
+
+        @user_count = accessible_users_by_role.count
+
+        accessible_users_by_role.paginate(page: @page, per_page: 5)
```
Comment (tofarr) [inline app/presenters/hub/dashboard/team_assignment_presenter.rb:24]: Should we do fetch request like we do for the Returns by Status module?

File: app/models/user.rb (hunk 1 above)
Comment (powersurge360) [reply app/models/user.rb:264]: I'm doing a similar look up in [my PR](https://github.com/codeforamerica/vita-min/pull/4664/files#diff-9802ca3c9c4cf89904fd44bc114e35ebdf2c5dd3d5b645491e2b253e1afef29bR256-R287), but for a different purpose, carry on 🙂

NIT, though: In most other places we use eg `SiteCoordinatorRole::TYPE`.

**Review by tofarr — APPROVED** [review]
Good work Em!

File: app/presenters/hub/dashboard/team_assignment_presenter.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+                                   elsif @current_user.org_lead?
+                                     organization = @current_user.role.organization
+                                     sites = @selected_model.instance_of?(Site) ? @selected_model : VitaPartner.sites.where(parent_organization: organization)
+                                     roles = OrganizationLeadRole.where(organization: organization) + SiteCoordinatorRole.assignable_to_sites(sites) + TeamMemberRole.assignable_to_sites(sites)
+                                     User.where(role: roles)
+                                   elsif @current_user.site_coordinator?
+                                     sites = @selected_model.instance_of?(Site) ? @selected_model : @current_user.role.sites
+                                     site_coordinators = User.where(role: SiteCoordinatorRole.assignable_to_sites(sites))
+                                     team_members = User.where(role: TeamMemberRole.assignable_to_sites(sites))
+                                     site_coordinators.or(team_members)
```
Comment (jenny-heath) [inline app/presenters/hub/dashboard/team_assignment_presenter.rb:31]:
```suggestion
                                     roles = SiteCoordinatorRole.assignable_to_sites(sites) + TeamMemberRole.assignable_to_sites(sites)
                                     User.where(role: roles)                                   
```

**Pull request merged**